lib/websocket/connection.js

```diff
@@ -233,15 +350,10 @@ function receiveData (ws) {
   const { [kResponse]: response } = ws
 
   response.socket.on('data', (chunk) => {
-    const opcode = chunk[0] & 0x0F
-    const fin = (chunk[0] & 0x80) !== 0
-    const rsv1 = chunk[0] & 0x40
-    const rsv2 = chunk[0] & 0x20
-    const rsv3 = chunk[0] & 0x10
-
+    const frame = WebsocketFrame.from(chunk)
```

Comment:
Keep in mind that there is no guarantee that `chunk` is a complete frame. It might be just the first few bytes of a frame, a slice in the middle of a frame, the tail of frame and the beginning of another, or multiple frames.

Reply:
Hi @lpinca, Thank you for your feedback. As was under the assumption that any websocket server will flush the socket buffer after each complete frame and we'll always get a chunk with a full frame. 


Do you know an easy way to spin up a ws server where I can test these cases? 
1- buffer with incomplete frame
2- buffer with multiple frames
3. etc..

Reply:
> I was under the impression that any websocket server will flush the socket buffer after each complete frame and we'll always get a chunk with a full frame.

This is outside of the control of any server. It depends on the OS, the protocol (TCP), the network, etc. When you call `socket.write(data)`, there is no guarantee that the other peer receives all the data as single TCP packet.

> Do you know an easy way to spin up a ws server where I can test these cases?

It is easier to do it without a real server. Take a look at https://github.com/websockets/ws/blob/8.11.0/lib/receiver.js and https://github.com/websockets/ws/blob/8.11.0/test/receiver.test.js.

Keep also in mind that the spec advertises support for the `permessage-deflate` extension, so if the server you are connecting to also supports `permessage-deflate`, you might receive frames whose payload is compressed. If decompression is slower than incoming data, you also have to handle backpressure.

Reply:
Thank you @lpinca